components/locale_core/src/extensions/unicode/attributes.rs

```diff
@@ -148,6 +148,29 @@ impl Attributes {
     {
         self.deref().iter().map(|t| t.as_str()).try_for_each(f)
     }
+
+    /// Extends the `Attributes` with values from another `Attributes`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use icu::locale::extensions::unicode::Attributes;
+    ///
+    /// let mut attrs: Attributes = "foobar-foobaz".parse().unwrap();
+    /// let attrs2: Attributes = "foobar-fooqux".parse().unwrap();
+    ///
+    /// attrs.extend(attrs2);
+    ///
+    /// assert_eq!(attrs, "foobar-foobaz-fooqux".parse().unwrap());
+    /// ```
+    #[cfg(feature = "alloc")]
+    pub fn extend(&mut self, other: Attributes) {
```

Comment:
issue: these are new public APIs, we should consider these carefully

cc @sffc @robertbastian

Reply:
Ah, thanks!